src/kbmod/search/raw_image.cpp

```diff
@@ -238,239 +187,123 @@ void RawImage::convolve_cpu(const PSF& psf) {
-    float d_amt = (d_px + 1.0 - dx) * (dy - d_py);
-
-    // make sure the right amount has been distributed
-    float diff = std::abs(a_amt + b_amt + c_amt + d_amt - 1.0);
-    if (diff > 0.01) std::cout << "warning: bilinear_interpSum == " << diff << "\n";
-    return {a_px, a_py, a_amt, b_px, b_py, b_amt, c_px, c_py, c_amt, d_px, d_py, d_amt};
-}
-
-void RawImage::add_pixel_interp(float x, float y, float value) {
-    // Interpolation values
-    std::vector<float> iv = bilinear_interp(x, y);
-
-    add_to_pixel(iv[0], iv[1], value * iv[2]);
-    add_to_pixel(iv[3], iv[4], value * iv[5]);
-    add_to_pixel(iv[6], iv[7], value * iv[8]);
-    add_to_pixel(iv[9], iv[10], value * iv[11]);
-}
-
-void RawImage::add_to_pixel(float fx, float fy, float value) {
-    assert(fx - floor(fx) == 0.0 && fy - floor(fy) == 0.0);
-    int x = static_cast<int>(fx);
-    int y = static_cast<int>(fy);
-    if (x >= 0 && x < width && y >= 0 && y < height) pixels[y * width + x] += value;
-}
-
-float RawImage::get_pixel_interp(float x, float y) const {
-    if ((x < 0.0 || y < 0.0) || (x > static_cast<float>(width) || y > static_cast<float>(height)))
-        return NO_DATA;
-    std::vector<float> iv = bilinear_interp(x, y);
-    float a = get_pixel(iv[0], iv[1]);
-    float b = get_pixel(iv[3], iv[4]);
-    float c = get_pixel(iv[6], iv[7]);
-    float d = get_pixel(iv[9], iv[10]);
-    float interpSum = 0.0;
-    float total = 0.0;
-    if (a != NO_DATA) {
-        interpSum += iv[2];
-        total += a * iv[2];
-    }
-    if (b != NO_DATA) {
-        interpSum += iv[5];
-        total += b * iv[5];
-    }
-    if (c != NO_DATA) {
-        interpSum += iv[8];
-        total += c * iv[8];
-    }
-    if (d != NO_DATA) {
-        interpSum += iv[11];
-        total += d * iv[11];
-    }
-    if (interpSum == 0.0) {
-        return NO_DATA;
-    } else {
-        return total / interpSum;
-    }
-}
-
-void RawImage::set_all_pix(float value) {
-    for (auto& p : pixels) p = value;
-}
-
-std::array<float, 2> RawImage::compute_bounds() const {
-    const int num_pixels = get_npixels();
-    float min_val = FLT_MAX;
-    float max_val = -FLT_MAX;
-    for (unsigned p = 0; p < num_pixels; ++p) {
-        if (pixels[p] != NO_DATA) {
-            min_val = std::min(min_val, pixels[p]);
-            max_val = std::max(max_val, pixels[p]);
-        }
-    }
-
-    // Assert that we have seen at least some valid data.
-    assert(max_val != -FLT_MAX);
-    assert(min_val != FLT_MAX);
-
-    // Set and return the result array.
-    std::array<float, 2> res;
-    res[0] = min_val;
-    res[1] = max_val;
-    return res;
-}
+void RawImage::set_all(float value) { image.setConstant(value); }
 
+// I kind of honestly forgot "PixelPos" was a class. If we don't need
+// the "furthest from the center" thing the same operation can be
+// replaced by
+// Eigen::Index, i, j;
+// array.maxCoeff(&i, &j)
+// Not sure I understand the condition of !furthest_from_center
```

Comment:
I'm all for getting rid of furthest_from_center

Reply:
Cleaned up the comments, leaving to remember to open an issue.